azurerm/internal/services/datafactory/tests/resource_arm_data_factory_dataset_json_test.go

```diff
@@ -0,0 +1,297 @@
+
+		resp, err := client.Get(ctx, resourceGroup, dataFactoryName, name, "")
+
+		if err != nil {
+			return nil
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("Data Factory Dataset HTTP still exists:\n%#v", resp.Properties)
+		}
+	}
+
+	return nil
+}
+
+func testAccAzureRMDataFactoryDatasetJSON_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_data_factory" "test" {
+  name                = "acctestdf%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_data_factory_linked_service_web" "test" {
+  name                = "acctestlsweb%d"
+  resource_group_name = azurerm_resource_group.test.name
+  data_factory_name   = azurerm_data_factory.test.name
+  authentication_type = "Anonymous"
+  url                 = "http://www.bing.com"
+}
+
+resource "azurerm_data_factory_dataset_json" "test" {
+  name                = "acctestds%d"
+  resource_group_name = azurerm_resource_group.test.name
+  data_factory_name   = azurerm_data_factory.test.name
+  linked_service_name = azurerm_data_factory_linked_service_web.test.name
+
+  http_server_location {
+    url      = "http://www.bing.com"
+    path     = "foo/bar/"
+    filename = "foo.json"
+  }
+
+  encoding = "UTF-8"
+}
+`, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger, data.RandomInteger)
+}
+
+func testAccAzureRMDataFactoryDatasetJSON_update1(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_data_factory" "test" {
+  name                = "acctestdf%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+}
+
+resource "azurerm_data_factory_linked_service_web" "test" {
+  name                = "acctestlsweb%d"
+  resource_group_name = azurerm_resource_group.test.name
+  data_factory_name   = azurerm_data_factory.test.name
+  authentication_type = "Anonymous"
+  url                 = "http://www.bing.com"
+}
+
+resource "azurerm_data_factory_dataset_json" "test" {
+  name                = "acctestds%d"
+  resource_group_name = azurerm_resource_group.test.name
+  data_factory_name   = azurerm_data_factory.test.name
+  linked_service_name = azurerm_data_factory_linked_service_web.test.name
+
+  http_server_location {
+    url      = "http://www.bing.com"
+    path     = "foo/bar/"
+    filename = "foo.json"
+  }
+
+  encoding = "UTF-8"
+
+  description = "test description"
+  annotations = ["test1", "test2", "test3"]
+  folder      = "testFolder"
+
+  parameters = {
+    foo = "test1"
+    bar = "test2"
```

Comment:
We should put capitals everywhere possible to ensure the API handles them correctly
```suggestion
    foo = "test1"
    Bar = "Test2"
```